internal/storage/payload.go

```diff
@@ -176,6 +178,18 @@ func (w *PayloadWriter) AddBoolToPayload(msgs []bool) error {
 	return HandleCStatus(&status, "AddBoolToPayload failed")
 }
 
+func (w *PayloadWriter) AddByteToPayload(msgs []byte) error {
+	length := len(msgs)
+	if length <= 0 {
+		return errors.New("can't add empty msgs into payload")
+	}
+	cMsgs := (*C.int8_t)(unsafe.Pointer(&msgs[0]))
+	cLength := C.int(length)
+
+	status := C.AddInt8ToPayload(w.payloadWriterPtr, cMsgs, cLength)
```

Comment:
Better to have another `AddUint8ToPayload` interface. I noticed we use `arrow::Int8Builder` to write data. Not sure if there are some issues we also use `arrow::Int8Builder` to store `byte(uint8_t)`.